eval.py

```diff
@@ -0,0 +1,176 @@
+import os
+
+import hydra
+import torch
+import torch.nn as nn
+from omegaconf import OmegaConf
+from torch.utils.data import DataLoader
+
+from train import val_epoch
+from data.batch import Batch
+from model import make_model
+from data.dataset import BinsToVelocityDataset
+from modules.encoderdecoder import subsequent_mask
+from modules.label_smoothing import LabelSmoothing
+
+
+@hydra.main(version_base=None, config_path="config", config_name="eval_conf")
+def main(cfg):
+    checkpoint = load_checkpoint(
+        run_name=cfg.run_name,
+        epoch=cfg.model_epoch,
+        device=cfg.device,
+    )
+    train_cfg = OmegaConf.create(checkpoint["cfg"])
+    val_data = BinsToVelocityDataset(
+        split="test",
+        n_dstart_bins=3,
+        n_velocity_bins=3,
+        n_duration_bins=3,
+        sequence_len=train_cfg.sequence_size,
+        device=cfg.device,
+    )
+
+    dataloader = DataLoader(val_data, batch_size=train_cfg.train.batch_size)
+
+    input_size = len(val_data.src_vocab)
+    output_size = len(val_data.tgt_vocab)
+
+    model = make_model(
+        input_size=input_size,
+        output_size=output_size,
+        n=train_cfg.model.n,
+        d_model=train_cfg.model.d_model,
+        d_ff=train_cfg.model.d_ff,
+        dropout=train_cfg.model.dropout,
+    )
+    model.to(cfg.device)
+    model.load_state_dict(checkpoint["model_state_dict"])
+
+    if cfg.tasks.val_epoch:
+        pad_idx = val_data.tgt_vocab.index("<blank>")
+
+        criterion = LabelSmoothing(
+            size=output_size,
+            padding_idx=pad_idx,
+            smoothing=train_cfg.train.label_smoothing,
+        )
+        criterion.to(cfg.device)
+
+        print("Evaluating model ...")
+        loss = val_epoch(
+            dataloader=dataloader,
+            model=model,
+            criterion=criterion,
+        )
+        print(f"Model loss:   {loss}")
+
+    if cfg.tasks.translation:
+        print("Checking model outputs ...")
+        translations = make_examples(
+            dataset=val_data,
+            model=model,
+            n_examples=cfg.n_examples,
+            random=cfg.random,
+        )
+
+        for translation in translations:
+            print("Source (Input)        : " + " ".join(translation["src"]))
+            print("Target (Ground Truth) : " + " ".join(translation["tgt"]))
+            print("Model Output               : " + " ".join(translation["out"]))
+
+
+def load_checkpoint(run_name: str, epoch: str = "final", device: str = "cpu"):
+    # find path with desired run
+    path = None
+    for file in os.listdir("models"):
+        if file.find(f"{run_name}-{epoch}") > 0:
+            path = file
+            break
+
+    if path is None:
+        print("no run with this id found")
+        return None
+
+    path = "models/" + path
+    checkpoint = torch.load(path, map_location=device)
+    return checkpoint
```

Comment:
```suggestion
    path = f"models/{run_name}.pt"
    checkpoint = torch.load(path, map_location=device)
    return checkpoint
```

I think there's no need to track per-epoch versions - can you change your checkpointing logic to only save the best performing version?

```python
best_loss = float("inf")

for epoch in epochs:
    test_loss = test_epoch(model)
    if test_loss < best_loss:
        save_checkpoint(model)
        best_loss = test_loss
```

This way you'll also save some disk space :)